pkg/templateservicebroker/brokerapi/templateservicebroker/provision.go

```diff
@@ -0,0 +1,186 @@
+package templateservicebroker
+
+import (
+	"net/http"
+	"reflect"
+
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	"github.com/openshift/origin/pkg/templateservicebroker"
+	"github.com/openshift/origin/pkg/templateservicebroker/api"
+	"github.com/openshift/origin/pkg/templateservicebroker/brokerapi"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/types"
+)
+
+func (b *Broker) templateFromUID(uid types.UID) (*templateapi.Template, error) {
+	// TODO: cache this?
```

Comment:
Does an informer imply a controller loop to keep it up to date? If so is it a permissible pattern to access such a cache from the apiserver?